enumerables.rb

```diff
@@ -0,0 +1,143 @@
+# Defining Enumerable
+module Enumerable
+  def my_each
+    pos = 0
+    return to_enum unless block_given?
+
+    loop do
+      break if pos >= length
+
+      yield self[pos]
+      pos += 1
+    end
+    self
+  end
+
+  # Defining my_each_with_index
+  def my_each_with_index
+    pos = 0
+    return to_enum unless block_given?
+
+    loop do
+      break if pos >= length
+
+      yield self[pos], pos
+      pos += 1
+    end
+    self
+  end
+
+  # Defining my_select
+  def my_select
+    return_arr = []
+    return to_enum unless block_given?
+
+    my_each do |element|
+      return_arr.push(element) if yield element
+    end
+
+    return_arr
+  end
+
+  # Defining my_all
+  def my_all?(arg = nil)
+    return_val = true
+    my_each do |element|
+      return_val = (block_given? && yield(element)) || element.poitive?(arg)
+      return_val = true?(element) if !block_given? && !arg
+      return false unless return_val
+    end
+    return_val
+  end
+
+  # Defining my_any?
+  def my_any?(arg = nil)
+    return_val = false
+    my_each do |element|
+      return_val = (block_given? && yield(element)) || element.poitive?(arg)
+      return_val = true?(element) if !block_given? && !arg
+      return true if return_val
+    end
+    return_val
+  end
+
+  # Defining my_none?
+  def my_none?(arg = nil)
+    return_val = true
+    my_each do |element|
+      return_val = (block_given? && yield(element)) || element.poitive?(arg)
+      return_val = true?(element) if !block_given? && !arg
+      return false if return_val
+    end
+    true
+  end
+
+  # Defining my_count
+  def my_count(arg = nil)
+    count_val = 0
+    my_each do |element|
```

Comment:
This method should return the number of items in the enumerator when no argument or block is given.
It should take a block as an argument. Right now it raises the error: ` wrong number of arguments (given 1, expected 0)` the same as the above methods.

Reply:
Added Multiple_els for the above.